### PR TITLE
Run kazel on the entire tree

### PR DESCRIPTION
**Pull request by ixdy**
**What this PR does / why we need it**: part of #47558: auto-generate `BUILD` files on the entire tree, since this is what `gazelle` does, and it'll make subsequent reviews easier if less is changing.

**Release note**:

```release-note
NONE
```
/assign
/release-note-none

### CODE REVIEW

**Comment by ixdy** [comment]
/unassign
/assign @mikedanese @spxtr 

Seems to work.

**Comment by ixdy** [comment]
(you might want to review individual commits and ignore the autogenerated one)

**Comment by ixdy** [comment]
This bazel failure is weird:
```
W0719 23:22:33.712] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/byte.go' was not created.
W0719 23:22:33.716] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/doc.go' was not created.
W0719 23:22:33.718] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/empty.go' was not created.
W0719 23:22:33.720] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/int.go' was not created.
W0719 23:22:33.721] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/int64.go' was not created.
W0719 23:22:33.723] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: output 'staging/src/k8s.io/apimachinery/pkg/util/sets/string.go' was not created.
W0719 23:22:33.730] ERROR: /workspace/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD:25:1: not all outputs were created or valid.
```

It's already the case in upstream/master that doing `bazel buildstaging/src/k8s.io/apimachinery/pkg/util/sets/...` fails in the same way, whereas `bazel build //vendor/k8s.io/apimachinery/pkg/util/sets/...` works as expected.
I'm not sure what changed here to break this.

**Comment by ixdy** [comment]
I have independently discovered that @jmillikin-stripe's #48565 fixes the bazel failure.

**Comment by ixdy** [comment]
/rtests
Tests are so flaky. At least bazel test is working now.

**Comment by ixdy** [comment]
/retest

this time with correct spelling

File: hack/BUILD [hunk 1]
```diff
@@ -47,3 +48,22 @@ test_suite(
         #        "verify-gofmt",
     ],
 )
+
+go_binary(
+    name = "hack",
```
Comment (ixdy) [inline hack/BUILD]: this one is probably better named `e2e` but I'm guessing gazel wouldn't support that.

File: staging/BUILD [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+licenses(["notice"])
+
+load(
+    "@io_bazel_rules_go//go:def.bzl",
+    "go_binary",
+    "go_library",
+)
+
+go_binary(
+    name = "staging",
```
Comment (ixdy) [inline staging/BUILD]: this one is also mis-named.

**Comment by ixdy** [comment]
other than the two `go_binary`s I mentioned, the autogenerated stuff looks as expected.

**Comment by spxtr** [comment]
Is this ready to go in?

**Comment by ixdy** [comment]
yes, I'm waiting for review and then I'll hunt down a TLA.

**Comment by ixdy** [comment]
(I just rebased + regenerated. hopefully it all still works. :)

**Comment by ixdy** [comment]
/assign @thockin 

can I get TLA for the BUILD file changes?

**Comment by ixdy** [comment]
@k8s-merge-robot wake up, kops is green

**Comment by ixdy** [comment]
Rebased and regenerated.

**Comment by spxtr** [comment]
[Are we really that flaky right now?](https://k8s-gubernator.appspot.com/pr/48969) Jeez.

**Comment by ixdy** [comment]
🎄🎄🎄

**Comment by ixdy** [comment]
(still need a re-LGTM)

**Comment by ixdy** [comment]
Rebased and regenerated *again*. ARGH.

**Comment by ixdy** [comment]
only change since last lgtm tag was regenerating the BUILD files. I've checked that commit manually and it looks as expected, so retagging so this can merge overnight before it breaks again.

**Pull request merged**